echopype/convert/parse_azfp.py

```diff
@@ -40,112 +131,119 @@ def get_value_by_tag_name(tag_name, element=0):
-            "VTX0": "VTX",
-            "BP": "BP",
-        }
-
         # Retrieve integer parameters from the xml file
-        for old_name, new_name in int_params.items():
+        for old_name, new_name in XML_INT_PARAMS.items():
             self.parameters[new_name] = int(get_value_by_tag_name(old_name))
         # Retrieve floating point parameters from the xml file
-        for param in float_params:
+        for param in XML_FLOAT_PARAMS:
             self.parameters[param] = float(get_value_by_tag_name(param))
         # Retrieve frequency dependent parameters from the xml file
-        for old_name, new_name in freq_params.items():
+        for old_name, new_name in XML_FREQ_PARAMS.items():
             self.parameters[new_name] = [
                 float(get_value_by_tag_name(old_name, ch))
                 for ch in range(self.parameters["num_freq"])
             ]
 
-    def parse_raw(self):
-        """Parse raw data file from AZFP echosounder.
+    def _compute_temperature(self, ping_num, is_valid):
+        """
+        Compute temperature in celsius.
 
         Parameters
         ----------
-        raw : list
-            raw filename
+        ping_num
+            ping number
+        is_valid
+            whether the associated parameters have valid values
         """
+        if not is_valid:
+            return np.nan
+
+        counts = self.unpacked_data["ancillary"][ping_num][4]
+        v_in = 2.5 * (counts / 65535)
+        R = (self.parameters["ka"] + self.parameters["kb"] * v_in) / (self.parameters["kc"] - v_in)
+
+        # fmt: off
+        T = 1 / (
+            self.parameters["A"]
+            + self.parameters["B"] * (np.log(R))
+            + self.parameters["C"] * (np.log(R) ** 3)
+        ) - 273
+        # fmt: on
+        return T
+
+    def _compute_tilt(self, ping_num, xy, is_valid):
+        """
+        Compute instrument tilt.
 
-        # Start of computation subfunctions
-        def compute_temp(counts):
-            """
-            Returns the temperature in celsius given from xml data
-            and the counts from ancillary
-            """
-            v_in = 2.5 * (counts / 65535)
-            R = (self.parameters["ka"] + self.parameters["kb"] * v_in) / (
-                self.parameters["kc"] - v_in
-            )
-            # fmt: off
-            T = 1 / (
-                self.parameters["A"]
-                + self.parameters["B"] * (math.log(R))
-                + self.parameters["C"] * (math.log(R) ** 3)
-            ) - 273
-            # fmt: on
-            return T
-
-        def compute_tilt(N, a, b, c, d):
+        Parameters
+        ----------
+        ping_num
+            ping number
+        xy
+            either "X" or "Y"
+        is_valid
+            whether the associated parameters have valid values
+        """
+        if not is_valid:
+            return np.nan
+        else:
+            N = self.unpacked_data["ancillary"][ping_num][0]
```

Comment:
Modifying the list index to be 0 or 1, per the original code.
```suggestion
            idx = 0 if xy == "X" else 1
            N = self.unpacked_data["ancillary"][ping_num][idx]
```

Reply:
Ah you're right, thanks for catching this!

Reply:
I thought I added it to the batch but apparently not. I'll commit it now.